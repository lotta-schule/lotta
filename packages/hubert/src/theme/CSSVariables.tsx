import { Theme } from '@lotta-schule/theme';
import * as React from 'react';
import { toCSSVariableName, toCSSVariableValue } from '../util';

export type CSSVariablesProps = {
  theme: Theme;
};

export const CSSVariables = React.memo(({ theme }: CSSVariablesProps) => {
  const cssVarsInit = Object.entries(theme)
    .map(
      ([key, val]) =>
        `    ${toCSSVariableName(key)}: ${toCSSVariableValue(val)};`
    )
    .join('\n');

  const cssContent = ':root {\n' + cssVarsInit + '\n}';

  return <style dangerouslySetInnerHTML={{ __html: cssContent }} />;
});
CSSVariables.displayName = 'CSSVariables';
