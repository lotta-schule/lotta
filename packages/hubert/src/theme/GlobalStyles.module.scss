:global {
  * {
    box-sizing: border-box;
  }

  html,
  body {
    margin: 0;
    font-family: var(--lotta-text-font-family);

    @media print {
      font-size: 8pt;
    }
  }

  body {
    margin: 0 auto;

    &::after {
      content: '';
      position: fixed;
      top: 0;
      height: 100vh; /* fix for mobile browser address bar appearing disappearing */
      left: 0;
      right: 0;
      z-index: -1;
      background-color: rgba(var(--lotta-page-background-color), 1);
      background-attachment: scroll;
      background-size: cover;
    }
  }

  ul {
    padding: 0;
    margin: 0;
    list-style: none;
  }

  a {
    color: inherit;
    text-decoration: inherit;
  }

  figure {
    margin: 0;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin-block-start: 0;
    margin-block-end: calc(0.5 * var(--lotta-spacing));
  }

  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.25em;
  }
  h4 {
    font-size: 1.2em;
  }
  h5,
  h6 {
    font-size: 1em;
  }

  fieldset {
    border: none;
    padding: 0;
  }
}
