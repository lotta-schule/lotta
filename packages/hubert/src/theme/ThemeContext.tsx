import * as React from 'react';
import { DefaultThemes, Theme } from '@lotta-schule/theme';

export const ThemeContext = React.createContext({
  theme: DefaultThemes.standard,
});

export interface ThemeProviderProps {
  theme: Theme;
  children?: React.ReactNode;
}

export const ThemeProvider = ({ children, ...props }: ThemeProviderProps) => (
  <ThemeContext.Provider value={props}>{children}</ThemeContext.Provider>
);

export const useTheme = () => React.useContext(ThemeContext).theme;
