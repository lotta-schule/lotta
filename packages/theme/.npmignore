# dotfiles
.*

node_modules

# We want the dist folder to be included in the repo
!dist

# typescript build artifacts
*.tsbuildinfo

# coverage directory
coverage

# test files
*.test.ts
*.test.tsx
jest.config.ts
jest.setup.ts

