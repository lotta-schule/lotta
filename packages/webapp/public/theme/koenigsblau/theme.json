{
  "pageBackgroundColor": "#cacdd7",
  "primaryColor": "#092f99",
  "navigationBackgroundColor": "#00b4ec",
  "navigationColor": "#475d68",
  "textColor": "#394151",
  "primaryContrastTextColor": "#475d68"
}
