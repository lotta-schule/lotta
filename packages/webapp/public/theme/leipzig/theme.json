{
  "pageBackgroundColor": "#e1ddd8",
  "primaryColor": "#1d4eab",
  "navigationBackgroundColor": "#f1aa00",
  "textColor": "rgb(5, 13, 36)",
  "navigationContrastTextColor": "#777e92",
  "primaryContrastTextColor": "#777e92"
}
