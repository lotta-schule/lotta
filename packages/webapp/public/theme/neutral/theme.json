{
  "pageBackgroundColor": "#e1ddd8",
  "primaryColor": "#8f9ba4",
  "navigationBackgroundColor": "#eef5f7",
  "textColor": "rgb(5, 13, 36)",
  "navigationContrastTextColor": "#53555f",
  "primaryContrastTextColor": "#ffffff",
  "errorColor": "#dc4c6a",
  "successTextColor": "#2f8c5b"
}
