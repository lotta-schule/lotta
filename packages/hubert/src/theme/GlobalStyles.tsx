import * as React from 'react';
import { Theme } from '@lotta-schule/theme';
import { CSSVariables } from './CSSVariables';
import { ExternalFont, FontLoader } from './FontLoader';

import './GlobalStyles.global.scss';

export type GlobalStylesProps = {
  theme: Theme;
  supportedFonts?: ExternalFont[];
};

export const GlobalStyles = React.memo(
  ({ theme, supportedFonts = [] }: GlobalStylesProps) => {
    return (
      <>
        <CSSVariables theme={theme} />
        <FontLoader theme={theme} supportedFonts={supportedFonts} />
      </>
    );
  }
);
GlobalStyles.displayName = 'GlobalHubertStyles';
