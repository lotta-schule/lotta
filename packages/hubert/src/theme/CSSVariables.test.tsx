import * as React from 'react';
import { CSSVariables } from './CSSVariables';
import { render } from '@testing-library/react';
import { DefaultThemes } from '@lotta-schule/theme';

describe('CSSVariables', () => {
  it('should output the correct css variable definitions for the default theme', () => {
    const screen = render(<CSSVariables theme={DefaultThemes['standard']} />);
    expect(screen.container.querySelector('style')).toMatchInlineSnapshot(`
      <style>
        :root {
          --lotta-primary-color: 255, 87, 34;
          --lotta-navigation-background-color: 51, 51, 51;
          --lotta-error-color: 255, 0, 0;
          --lotta-success-color: 10, 82, 37;
          --lotta-navigation-color: 51, 51, 51;
          --lotta-disabled-color: 97, 97, 97;
          --lotta-text-color: 33, 33, 33;
          --lotta-label-text-color: 158, 158, 158;
          --lotta-contrast-text-color: 255, 255, 255;
          --lotta-box-background-color: 255, 255, 255;
          --lotta-page-background-color: 202, 205, 215;
          --lotta-divider-color: 224, 224, 224;
          --lotta-highlight-color: 224, 224, 224;
          --lotta-banner-background-color: 54, 123, 240;
          --lotta-accent-grey-color: 227, 227, 227;
          --lotta-spacing: 8px;
          --lotta-border-radius: 4px;
          --lotta-text-font-family: Muli;
          --lotta-title-font-family: 'Schoolbell', cursive;
      }
      </style>
    `);
  });
});
