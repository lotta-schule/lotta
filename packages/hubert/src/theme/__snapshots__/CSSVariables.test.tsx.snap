// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSSVariables > should output the correct css variable definitions for the default theme 1`] = `
<style
  data-hubert-css-variables-def="true"
>
  :root {
    --lotta-primary-color: 255, 87, 34;
    --lotta-navigation-background-color: 51, 51, 51;
    --lotta-navigation-contrast-text-color: 234, 234, 234;
    --lotta-error-color: 255, 0, 0;
    --lotta-success-color: 10, 82, 37;
    --lotta-disabled-color: 97, 97, 97;
    --lotta-text-color: 33, 33, 33;
    --lotta-label-text-color: 158, 158, 158;
    --lotta-primary-contrast-text-color: 255, 255, 255;
    --lotta-box-background-color: 255, 255, 255;
    --lotta-page-background-color: 202, 205, 215;
    --lotta-divider-color: 224, 224, 224;
    --lotta-highlight-color: 224, 224, 224;
    --lotta-banner-background-color: 54, 123, 240;
    --lotta-accent-grey-color: 227, 227, 227;
    --lotta-spacing: 8px;
    --lotta-border-radius: 4px;
    --lotta-text-font-family: Muli;
    --lotta-title-font-family: 'Schoolbell', cursive;
}
</style>
`;
